DOCS-9859

domalessi opened the pull request:
<!-- *Note: Please remember to review the Datadog Documentation [Contribution Guidelines](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md) if you have not yet done so.* -->

### What does this PR do? What is the motivation?
https://datadoghq.atlassian.net/browse/DOCS-9859
Adding use case docs to Service Catalog docs.

### Merge instructions

<!-- 
If you're waiting for a release or there are other considerations that you want us to be aware of, list them here. 
If the PR is ready to be merged once it receives the required reviews, check the box below after you've created the PR.
-->

Merge readiness:
- [ ] Ready for merge

Merge queue is enabled in this repo. To have it automatically merged after it receives the required reviews, create the PR (from a branch that follows the `<yourname>/description` naming convention) and then add the following PR comment:

```
/merge
```

### Additional notes
<!-- Anything else we should know when reviewing?-->

<!-- Previewing the PR: Assuming you are a Datadog employee and named your branch `<yourname>/<description>`, a preview build will run and links to the preview output will be auto-generated and posted in the PR comments. The links will 404 until the preview build is finished running. -->

buraizu commented:
Created DOCS-9999 to track

janine-c commented on content/en/service_catalog/use_cases/production_readiness.md:
```diff
@@ -0,0 +1,65 @@
+- Detect gaps such as missing SLOs or monitors, or unowned services.
+- Enforce tagging best practices and verify configurations for enabling cross-telemetry insights.
+
+Open the Setup Guidance tab for a given service to check whether you've completed the necessary setup to leverage Datadog features like Monitors, SLOs, and Error Tracking. You can also check if your service is properly configured to collect key telemetry data, like tracing and logs.
```
I can imagine new users getting lost looking for this tab too. Could you do something like provide a link into the app, or give some more context in your screenshot, so they don't get confused?

janine-c commented on content/en/service_catalog/use_cases/production_readiness.md:
```diff
@@ -0,0 +1,65 @@
+
+In the Costs tab, powered by [Cloud Cost Management][2], you can identify inefficiencies, cost-saving opportunities, and trends over time. With cost data and service metrics in one place, you can understand how engineering changes impact overall cloud spend.
+
+For more information on a service’s cost changes, click on the service row to open a detailed side panel, or go to the Cloud Costs page by clicking “View in Analytics.”
```
```suggestion
For more information on a service’s cost changes, click on the service row to open a detailed side panel, or go to the Cloud Costs page by clicking **View in Analytics**.
```

janine-c commented on content/en/service_catalog/use_cases/production_readiness.md:
```diff
@@ -0,0 +1,65 @@
+
+### Use the Security tab to address vulnerabilities
+
+Use the Security tab in Service Catalog to identify and remediate vulnerabilities in service dependencies. This view also reveals which services are most targeted by attacks and exposed to severe threats.
```
The screenshot kind of implies this, but it might be worth clarifying that the Security tab isn't just hanging around in Service/Software Catalog, and that you have to click on a service first

janine-c reviewed (commented):
Looks awesome, Dom! I made some readability comments, absolutely none of which are show-stoppers. Let me know how else I can support you getting this one out the door!

domalessi commented on content/en/service_catalog/use_cases/pipeline_visibility.md:
```diff
@@ -0,0 +1,52 @@
+- Monitor the performance of CI pipelines related to your services.
+- Identify security and code quality issues from [Static Analysis][1].
+- Troubleshoot pre-production slowdowns and failures.
+- Track Change Lead Time by integrating with [DORA Metrics][2].
```
Change Lead Time is capitalized in the DORA Metrics docs, so I left it that way to be consistent!

domalessi commented on content/en/service_catalog/use_cases/incident_response.md:
```diff
@@ -0,0 +1,43 @@
+
+Service Catalog also integrates with [Datadog Incident Management][1] and [PagerDuty][2], allowing you to view related incidents in the Reliability tab on the Service Details page.
+
+**Note**: Datadog Incidents automatically link to Service Catalog; apply appropriate `SERVICE` tags to an incident to ensure that a service’s incident data are accurate. The PagerDuty integration must be manually set up to integrate with incident information in Service Catalog. 
```
That's fair... I was trying to differentiate between steps you need to take for DD Incidents vs PagerDuty Integration. I'll take another stab at it!

domalessi commented on content/en/service_catalog/use_cases/pipeline_visibility.md:
```diff
@@ -0,0 +1,52 @@
+
+To add or remove a pipeline associated with your service:
+
+1. Click Edit Metadata in your service page, and go to Software Delivery. **Note**: This is only available for [Service Catalog schema v2.2][3].
```
You are so right! This is very confusing. I'll update.

domalessi commented:
@janine-c Thank you so much for the review, Janine! I've updated the PR to address your feedback and improve clarity in some sections. I've also updated everything to align with the new Software Catalog name. Mind doing one more quick glance-through?

domalessi commented on content/en/software_catalog/use_cases/production_readiness.md:
```diff
@@ -0,0 +1,64 @@
+
+Use the Security tab in Software Catalog to identify and remediate vulnerabilities in service dependencies. This view also reveals which services are most targeted by attacks and exposed to severe threats.
+
+To explore security details for a particular service, click on a service in Software Catalog to open the service side panel, and find the Security tab. 
+
+{{< img src="tracing/software_catalog/production-readiness-security.png" alt="The security tab for a service, showing details about vulnerability risk and attack exposure" style="width:100%;" >}}
+
+You can also explore the Security tab on the Service Page, which can be found by hovering over a service name and selecting **Service Page**, or using the **Service Page** button in the top right corner of the side panel.
```
@janine-c In particular, would love your thoughts on lines 44-48 and 56. Is this more clear? Or is my attempt at clarity/adding more detail actually just more confusing? 

I'm toying with adding additional images or maybe even a little screengrab vid but I don't want to overcomplicate either.

janine-c replied:
Definitely a step in the right direction! I'm wondering if adding some bullets/other visual formatting might make it easier to scan through as a finishing touch to limit the cognitive load it takes to comprehend that these are various ways of viewing the same information. But it's not a blocker by any means! And I definitely agree that you don't want to overcomplicate things here - I think you struck a good balance there.

janine-c commented on content/en/software_catalog/use_cases/pipeline_visibility.md:
```diff
@@ -0,0 +1,60 @@
+
+To add or remove a pipeline associated with your service:
+
+1. Click on your service in [Software Catalog][4] to open the service side panel, and select one of the edit options for Entity Metadata. . 
```
Are the edit options you're talking about on the Ownership tab? I was clicking around the UI and it defaulted to the Performance tab, so I got really lost looking for the edit options!

janine-c commented on content/en/software_catalog/use_cases/pipeline_visibility.md:
```diff
@@ -0,0 +1,60 @@
+
+2. Then, you can either:
+
+   - Find the Software Delivery section, then search for and select the pipelines you want to associate.
```
Before looking in the UI, I thought this meant associating pipelines with each other, as opposed to pipelines with the service.
```suggestion
   - Find the Software Delivery section, then search for and select the pipelines you want to associate with the service.
```

janine-c commented on content/en/software_catalog/use_cases/pipeline_visibility.md:
```diff
@@ -0,0 +1,60 @@
+
+      {{< img src="tracing/software_catalog/pipeline-visibility-update-metadata.png" alt="The configuration page for updating service metadata, featuring the Software Delivery field for adding and removing related pipelines" style="width:100%;" >}}
+
+   - Manually add a pipeline fingerprint to the service metadata. To find a pipeline's fingerprint, go to the [Pipelines][5] page, click on the pipeline, and select the gear icon.
```
This tells you how to find the fingerprint, but not what to do with it after. I think I'd put this step above the previous one, and frame it as an optional step you can take before selecting pipelines. I think we could probably frame searching/selecting a pipeline and pasting in a fingerprint as the same step? But I leave the nuances up to you and your expertise!

amlakhanpal replied:
The answer is both! You can either add it through the UI or just when you're creating the YAML file as ci-pipeline-fingerprints. Example here: https://docs.datadoghq.com/software_catalog/service_definitions/v2-2/

janine-c commented on content/en/software_catalog/use_cases/pipeline_visibility.md:
```diff
@@ -0,0 +1,60 @@
+
+      {{< img src="tracing/software_catalog/pipeline-visibility-pipeline-fingerprint.png" alt="An example of a pipeline fingerprint" style="width:100%;" >}}
+
+For more details on CI status and static analysis violations for a particular service, click on the service and navigate to the Delivery tab to review the status of each pipeline and rule violation.
```
```suggestion
For more details on CI status and static analysis violations, click on the service and navigate to the **Delivery** tab to review the status of each pipeline and rule violation.
```

janine-c commented on content/en/software_catalog/use_cases/production_readiness.md:
```diff
@@ -0,0 +1,64 @@
+- Detect gaps such as missing SLOs or monitors, or unowned services.
+- Enforce tagging best practices and verify configurations for enabling cross-telemetry insights.
+
+Open the Setup Guidance tab for a given service to check whether you've completed the necessary setup to leverage Datadog features like Monitors, SLOs, and Error Tracking. You can also check if your service is properly configured to collect key telemetry data, like tracing and logs.
```
I still think that a new user would have trouble finding this tab. Can you add a link and tell them to click the service to open the side panel, where they can then find the **Setup Guidance** tab?

janine-c reviewed (commented):
Beeyootiful! Just a couple mall comments but we should be good to go after these 💪🏻

domalessi commented:
Thank you so much for the additional review + feedback, @janine-c !!! Waiting on my PMs to follow up on one of your comments/questions. 

Would also love a quick check of this guy: https://docs-staging.datadoghq.com/domalessi/docs-9859/software_catalog/use_cases/production_readiness. Under the "Use the Security tab to address vulnerabilities" and "Use the Costs tab to optimize spending" sections, I broke out some of the text into bullets, per your suggestion. I also added one image that I hope makes things clearer instead of bloated. 

Otherwise, I think we're mostly set!

janine-c commented:
Those retooled sections look beautiful, @domalessi! The formatting and the extra screenshot really help demystify the UI, which can have so much going on in it that it's hard to pick out the important parts, especially for new users. It's so much easier to follow what you're saying now ✨ Thanks for putting the time into doing that!

domalessi commented:
@janine-c Pushed a few more small changes to incorporate Amrita's note about pipeline fingerprints. I think just glance through the Pipeline Visibility page and then we should be good to go?

janine-c reviewed (approved):
Amazing!! Thanks so much for your investigative work in answering my questions, Dom!! Can't wait to see this live 🎉

domalessi commented:
/merge
